### PR TITLE
[RFR] Put take

### PR DESCRIPTION
**Pull request by ThieryMichel**
follow #9

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]
Can you split this in at least two pr:
- one for refactoring
- another for the new effects

It's currently hard to review and we shouldn't mix refactoring and new features

File: README.md [hunk 1]
```diff
@@ -76,10 +76,33 @@ Calls a continuation passing style function.
...
+```
+
+### take
+Wait for event sent by put and return its payload
+```js
+const payload = yield take('my event'); // { payload: data }
+```
+
+### spawn
+Launch another sg generator, but do not wait for it to end. Returning a function that will resolve when the generator is done instead.
```
Comment (djhi) [inline README.md]: s/end. Returning/end, returning/

File: src/effects/fork.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import sg from '../sg';
+import createEffect from './createEffect';
+
+export const handleForkEffect = sgImpl => ([callable, ...args], parentEmitter, emitter) => {
+    const promise = sgImpl(callable, emitter)(...args);
```
Comment (djhi) [inline src/effects/fork.js:5]: It doesn't check for callable to be a generator as done in [spawn](https://github.com/marmelab/sg/pull/8/files#diff-0e50efdcb840bb7943ea324e136720f6R6) ?

Comment (ThieryMichel) [reply src/effects/fork.js:5]: I'ts spawn that should not verify, since sg does it.

**Review by djhi — APPROVED** [review]

**Pull request merged**